web/src/dialog_widget.ts

```diff
@@ -229,7 +229,7 @@ export function launch(conf: DialogWidgetConfig): void {
 export function submit_api_request(
     request_method: AjaxRequestHandler,
     url: string,
-    data: Parameters<AjaxRequestHandler>[0]["data"] = {},
+    data: Omit<Parameters<AjaxRequestHandler>[0]["data"], "undefined"> | PatchRequestData = {},
```

Comment:
Two cases:

1. When data is passed in function call : `data` can never be `undefined`.
2. When data is not passed: `data` can be `undefined` . But I have added an empty `{}` in those function call. This is also consistent with how many other module call this function.

Hence `Omit`ing  `undefined` is justified IMO. Also this is needed to satisfy `channel.patch' type definition.